services/upload/src/main/java/app/coronawarn/server/services/federation/upload/payload/AllowedPropertiesMap.java

```diff
@@ -0,0 +1,55 @@
+/*-
+ * ---license-start
+ * Corona-Warn-App
+ * ---
+ * Copyright (C) 2020 SAP SE and all other contributors
+ * ---
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ---license-end
+ */
+
+package app.coronawarn.server.services.federation.upload.payload;
+
+import app.coronawarn.server.common.protocols.external.exposurenotification.ReportType;
+import app.coronawarn.server.services.federation.upload.config.UploadServiceConfig;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AllowedPropertiesMap {
```

Comment:
Just a suggestion that this property defaulting behaviour could be moved "closer" to the configuration data. The EFgsTransmission object could return the detault for a property if its flags allow it.